woocommerce/woocommerce
# Add action hooks to WooCommerce email trigger function. Resolves #24641

### All Submissions:


### Changes proposed in this Pull Request:

Adds an action hook to the email trigger. This allows developers to run their own code after the sending of a WooCommerce email is triggered. Action hook includes send result (true / false) and the email object. 

Sample use case: Allows inserting a note into order when an email is sent so customer service can easily see which emails have been sent by the plugin to the customer. 

Closes #24641

### How to test the changes in this Pull Request:

1. Add action to the theme's functions.php file. A simple test case would be to write to debug log when email is being sent:

add_action('woocommerce_email_trigger_new_order_order', 'agentur_neworder', 20, 2 );

function agentur_neworder($result, $email) {
	error_log('NEW ORDER RESULT: ' . $result);
	error_log( print_r($email, true) );
}

2. Place a new order
3. Check debug log. If successful it will contain the result note and email object. 

### Other information:


### Changelog entry

> New action hook for email trigger function which allows developers to run their own code after Woo emails are triggered.
